Pylint: Fix `consider-using-enumerate` issues.

## 💡 Description
Update the codebase to use `enumerate` instead of `range` and `len` together for looping through lists and arrays. `pylint` report is attached (`pylint . --disable=all --enable=consider-using-enumerate`):
```
************* Module pds.naif_pds4_bundler.classes.setup
pds/naif_pds4_bundler/classes/setup.py:147:12: C0200: Consider using enumerate instead of iterating with range and len (consider-using-enumerate)
pds/naif_pds4_bundler/classes/setup.py:512:8: C0200: Consider using enumerate instead of iterating with range and len (consider-using-enumerate)
************* Module pds.naif_pds4_bundler.classes.label.label
pds/naif_pds4_bundler/classes/label/label.py:53:20: C0200: Consider using enumerate instead of iterating with range and len (consider-using-enumerate)
pds/naif_pds4_bundler/classes/label/label.py:73:20: C0200: Consider using enumerate instead of iterating with range and len (consider-using-enumerate)
************* Module pds.naif_pds4_bundler.classes.product.product_metakernel
pds/naif_pds4_bundler/classes/product/product_metakernel.py:127:16: C0200: Consider using enumerate instead of iterating with range and len (consider-using-enumerate)
************* Module pds.naif_pds4_bundler.utils.files
pds/naif_pds4_bundler/utils/files.py:672:4: C0200: Consider using enumerate instead of iterating with range and len (consider-using-enumerate)
pds/naif_pds4_bundler/utils/files.py:692:4: C0200: Consider using enumerate instead of iterating with range and len (consider-using-enumerate)
```

--- a/src/pds/naif_pds4_bundler/classes/label/label.py
+++ b/src/pds/naif_pds4_bundler/classes/label/label.py
@@ -50,7 +50,7 @@ class PDSLabel:
                     )
                 else:
                     missions_text = f"{setup.mission_name}, "
-                    for i in range(len(setup.secondary_missions)):
+                    for i, _ in enumerate(setup.secondary_missions):
                         if i == len(setup.secondary_missions) - 1:
                             missions_text += f"and {setup.secondary_missions[i]}"
                         else:
@@ -70,7 +70,7 @@ class PDSLabel:
                     )
                 else:
                     observers_text = f"{setup.observer}, "
-                    for i in range(len(setup.secondary_observers)):
+                    for i, _ in enumerate(setup.secondary_observers):
                         if i == len(setup.secondary_observers) - 1:
                             observers_text += f"and {setup.secondary_observers[i]}"
                         else:

--- a/src/pds/naif_pds4_bundler/classes/label/label.py
+++ b/src/pds/naif_pds4_bundler/classes/label/label.py
@@ -50,11 +50,11 @@ class PDSLabel:
                     )
                 else:
                     missions_text = f"{setup.mission_name}, "
-                    for i, _ in enumerate(setup.secondary_missions):
+                    for i, sm_name in enumerate(setup.secondary_missions):
                         if i == len(setup.secondary_missions) - 1:
-                            missions_text += f"and {setup.secondary_missions[i]}"
+                            missions_text += f"and {sm_name}"
                         else:
-                            missions_text += f"{setup.secondary_missions[i]}, "
+                            missions_text += f"{sm_name}, "
 
                 self.PDS4_MISSION_NAME = f"{missions_text}"
             else:
@@ -70,11 +70,11 @@ class PDSLabel:
                     )
                 else:
                     observers_text = f"{setup.observer}, "
-                    for i, _ in enumerate(setup.secondary_observers):
+                    for i, so_name in enumerate(setup.secondary_observers):
                         if i == len(setup.secondary_observers) - 1:
-                            observers_text += f"and {setup.secondary_observers[i]}"
+                            observers_text += f"and {so_name}"
                         else:
-                            observers_text += f"{setup.secondary_observers[i]}, "
+                            observers_text += f"{so_name}, "
 
                 self.PDS4_OBSERVER_NAME = f"{observers_text} spacecraft and their"
             else:

--- a/src/pds/naif_pds4_bundler/classes/product/product_metakernel.py
+++ b/src/pds/naif_pds4_bundler/classes/product/product_metakernel.py
@@ -124,11 +124,11 @@ class MetaKernelProduct(Product):
                 )
             else:
                 missions_text = f"{self.setup.mission_name}, "
-                for i, _ in enumerate(self.setup.secondary_missions):
-                    if i == len(self.setup.secondary_missions) - 1:
-                        missions_text += f"and {self.setup.secondary_missions[i]}"
+                for i, sm_name in enumerate(self.setup.secondary_missions):
+                    if i == len(sm_name) - 1:
+                        missions_text += f"and {sm_name}"
                     else:
-                        missions_text += f"{self.setup.secondary_missions[i]}, "
+                        missions_text += f"{sm_name}, "
 
             self.PDS4_MISSION_NAME = f"{missions_text}"
         else:

--- a/src/pds/naif_pds4_bundler/classes/product/product_metakernel.py
+++ b/src/pds/naif_pds4_bundler/classes/product/product_metakernel.py
@@ -124,7 +124,7 @@ class MetaKernelProduct(Product):
                 )
             else:
                 missions_text = f"{self.setup.mission_name}, "
-                for i in range(len(self.setup.secondary_missions)):
+                for i, _ in enumerate(self.setup.secondary_missions):
                     if i == len(self.setup.secondary_missions) - 1:
                         missions_text += f"and {self.setup.secondary_missions[i]}"
                     else:

--- a/src/pds/naif_pds4_bundler/classes/setup.py
+++ b/src/pds/naif_pds4_bundler/classes/setup.py
@@ -144,7 +144,7 @@ class Setup:
         # dictionaries.
         #
         if hasattr(self, "mk"):
-            for i in range(len(self.mk)):
+            for i, _ in enumerate(self.mk):
                 if isinstance(self.mk[i]["name"], dict):
                     self.mk[i]["name"] = [self.mk[i]["name"]]
 
@@ -509,7 +509,7 @@ class Setup:
         #
         # There might be more than one kernel directory
         #
-        for i in range(len(self.kernels_directory)):
+        for i, _ in enumerate(self.kernels_directory):
             if os.path.isdir(cwd + os.sep + self.kernels_directory[i]):
                 self.kernels_directory[i] = cwd + os.sep + self.kernels_directory[i]
             if not os.path.isdir(self.kernels_directory[i]):

--- a/src/pds/naif_pds4_bundler/classes/setup.py
+++ b/src/pds/naif_pds4_bundler/classes/setup.py
@@ -144,9 +144,9 @@ class Setup:
         # dictionaries.
         #
         if hasattr(self, "mk"):
-            for i, _ in enumerate(self.mk):
-                if isinstance(self.mk[i]["name"], dict):
-                    self.mk[i]["name"] = [self.mk[i]["name"]]
+            for i, mk_name in enumerate(self.mk):
+                if isinstance(mk_name["name"], dict):
+                    mk_name["name"] = [mk_name["name"]]
 
         #
         # Meta-kernel configuration; if there is one coverage kernel, convert
@@ -509,11 +509,11 @@ class Setup:
         #
         # There might be more than one kernel directory
         #
-        for i, _ in enumerate(self.kernels_directory):
-            if os.path.isdir(cwd + os.sep + self.kernels_directory[i]):
-                self.kernels_directory[i] = cwd + os.sep + self.kernels_directory[i]
+        for i, kd_name in enumerate(self.kernels_directory):
+            if os.path.isdir(cwd + os.sep + kd_name):
+                self.kernels_directory[i] = cwd + os.sep + kd_name
             if not os.path.isdir(self.kernels_directory[i]):
-                handle_npb_error(f"Directory does not exist: {self.kernels_directory[i]}.")
+                handle_npb_error(f"Directory does not exist: {kd_name}.")
 
         os.chdir(cwd)
 

--- a/src/pds/naif_pds4_bundler/utils/files.py
+++ b/src/pds/naif_pds4_bundler/utils/files.py
@@ -669,17 +669,17 @@ def match_patterns(name, name_w_pattern, patterns):
     value = ""
     value_bool = False
 
-    for i, _ in enumerate(name_check):
-        if (name_check[i] == name[i]) and (not value_bool):
+    for i, n in enumerate(name_check):
+        if (n == name[i]) and (not value_bool):
             continue
-        if (name_check[i] == name[i]) and value_bool:
+        if (n == name[i]) and value_bool:
             value_bool = False
             values_list.append(value)
             value = ""
-        elif (name_check[i] == "$") and (not value_bool):
+        elif (n == "$") and (not value_bool):
             value_bool = True
             value += name[i]
-        elif (name_check[i] == "$") and value_bool:
+        elif (n == "$") and value_bool:
             value += name[i]
         else:
             raise

--- a/src/pds/naif_pds4_bundler/utils/files.py
+++ b/src/pds/naif_pds4_bundler/utils/files.py
@@ -669,7 +669,7 @@ def match_patterns(name, name_w_pattern, patterns):
     value = ""
     value_bool = False
 
-    for i in range(len(name_check)):
+    for i, _ in enumerate(name_check):
         if (name_check[i] == name[i]) and (not value_bool):
             continue
         if (name_check[i] == name[i]) and value_bool:
@@ -689,7 +689,7 @@ def match_patterns(name, name_w_pattern, patterns):
     # patterns.
     #
     values = {}
-    for i in range(len(values_list)):
+    for i, _ in enumerate(values_list):
         values[pattern_name_order[i]] = values_list[i]
 
     return values

--- a/src/pds/naif_pds4_bundler/utils/files.py
+++ b/src/pds/naif_pds4_bundler/utils/files.py
@@ -689,8 +689,8 @@ def match_patterns(name, name_w_pattern, patterns):
     # patterns.
     #
     values = {}
-    for i, _ in enumerate(values_list):
-        values[pattern_name_order[i]] = values_list[i]
+    for i, name in enumerate(values_list):
+        values[pattern_name_order[i]] = name
 
     return values
 

